lib/containers/common.pm

```diff
@@ -25,13 +26,17 @@ sub is_unreleased_sle {
 }
 
 sub activate_containers_module {
-    my ($running_version, $sp, $host_distri) = get_os_release;
-    my $suseconnect = script_output_retry('SUSEConnect --status-text', timeout => 240, retry => 3, delay => 60);
-    if ($suseconnect !~ m/Containers/) {
-        $running_version eq '12' ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
-        $suseconnect = script_output_retry('SUSEConnect --status-text', timeout => 240, retry => 3, delay => 60);
+    my $json = Mojo::JSON::decode_json(script_output_retry('SUSEConnect -s', timeout => 240, retry => 3, delay => 60));
+    my ($not_registered, $version, $arch);
+    foreach (@$json) {
+        if ($_->{identifier} =~ 'container' && $_->{status} =~ 'Not Registered') {
```

Comment:
I'm nitpicking but IMHO it's better to do two things here:

1. Check for the explicit module name `sle-module-containers`, same as in line 38 where it is also used explicitly
2. Instead of checking for "Not Registered", check for not "Registered" - this check is broader as it would recognise any new status that is not explicitly "Registered" e.g. "Expired"
```suggestion
        if ($_->{identifier} =~ 'sle-module-containers' && $_->{status} !~ 'Registered') {
```

Reply:
> Check for the explicit module name sle-module-containers, same as in line 38 where it is also used explicitly

+1 

> Instead of checking for "Not Registered", check for not "Registered" - this check is broader as it would recognise any new status that is not explicitly "Registered" e.g. "Expired"

not sure here , if we will grep just for `Registered` , `Not Registered` will also match right ?